pyaedt/modules/PostProcessor.py

```diff
@@ -601,7 +601,351 @@ def surfacePlotInstruction(self):
+            New name of the plot.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        self.oreportsetup.RenameReport(PlotName, newname)
+        return True
+
+    @aedt_exception_handler
+    def get_report_data(self, expression="dB(S(1,1))", setup_sweep_name='', domain='Sweep', families_dict=None,
+                        report_input_type=None):
+        """Generate report data using the :func:`GetSolutionDataPerVariation` method.
+
+        This method returns the data object and the arrays ``solData`` and
+        ``FreqVals``.
+
+        Parameters
+        ----------
+        expression : str or list
+            One or more formulas to add to the report. The default is
+            ``"dB(S(1,1))"``.
+        setup_sweep_name : str, optional
+            Name of the setup for computing the report. The
+            default is ``""``, in which case the nominal sweep is
+            used.
+        domain : str or list, optional
+            Context type. The options are ``"Sweep"`` or
+            ``"Time"``. The default is ``"Sweep".``
+        families_dict : dict, optional
+            Dictionary of all families including the primary
+            sweep. The default is ``{"Freq": ["All"]}``.
+        report_input_type :  str
+            Type of input data for the report.
+
+        Returns
+        -------
+        :class:`pyaedt.modules.PostProcessor.SolutionData`
+
+        Examples
+        --------
+        Generate a report with the default sweep and default variation.
+
+        >>> hfss = HFSS()
+        >>> hfss.post.get_report_data("S(1,1)")
+
+        >>> m3d = Maxwell3D()
+        >>> m3d.post.get_report_data("SurfaceLoss")   # Eddy Current examples
+        >>> m3d.post.get_report_data("Wind(LoadA,LaodA)")    # TransientAnalsysis
+
+        """
+        if self.post_solution_type in ["3DLayout", "NexximLNA", "NexximTransient"]:
+            if domain == "Sweep":
+                did = 3
+            else:
+                did = 1
+            ctxt = ["NAME:Context", "SimValueContext:=",
+                    [did, 0, 2, 0, False, False, -1, 1, 0, 1, 1, "", 0, 0, "IDIID", False, "1"]]
+        elif type(domain) is list:
+            ctxt = domain
+        else:
+            ctxt = ["Domain:=", domain]
+
+        if type(expression) is not list:
+            expression = [expression]
+        if not setup_sweep_name:
+            setup_sweep_name = self._parent.nominal_sweep
+
+        if not report_input_type:
+            report_input_type = report_type[self.post_solution_type]
+
+        if families_dict is None:
+            families_dict = {"Freq": ["All"]}
+
+        solution_data = self.get_solution_data_per_variation(report_input_type, setup_sweep_name, ctxt, families_dict,
+                                                             expression)
+
+        if not solution_data:
+            print("No Data Available. Check inputs")
```

Comment:
Errors should be raised, not printed.  If we want a soft failure, use the `warnings.warn`.